lucene/test-framework/src/java/org/apache/lucene/index/BasePointsFormatTestCase.java

```diff
@@ -1174,8 +1174,11 @@ public void testMixedSchema() throws Exception {
 
     Document doc = new Document();
     doc.add(new IntPoint("id", 0));
-    w.addDocument(doc);
-    // now we write another segment where the id field does have points:
+    IllegalArgumentException ex =
+        expectThrows(IllegalArgumentException.class, () -> w.addDocument(doc));
+    assertEquals(
+        "cannot change field \"id\" from index options=DOCS to inconsistent index options=NONE",
+        ex.getMessage());
```

Comment:
I think we should refactor or drop this test, as it is not testing the points format now, but IndexingChain/FieldsInfos' logic. Maybe we could rename the test `testMergeMissing` and configure the first segment to not have the `id` field at all.

Reply:
The test refactored [here](https://github.com/apache/lucene/pull/11/commits/0fe3493110ac2a5f750ad41f732436daff6c69f5#diff-974afcef788664579ec220b758dd40bf67d8e72f3fb214e52016b8465fb54289L1161)